src/Strategy/GreedyCacheStrategy.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Kevinrob\GuzzleCache\Strategy;
+
+use Kevinrob\GuzzleCache\CacheEntry;
+use Kevinrob\GuzzleCache\Storage\CacheStorageInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * This strategy represent a "greedy" HTTP client.
+ *
+ * It can be used to cache responses in spite of any cache related response headers,
+ * but it SHOULDN'T be used unless absolutely necessary, e.g. when accessing
+ * badly designed APIs without Cache control.
+ *
+ * Obviously, this follows no RFC :(.
+ */
+class GreedyCacheStrategy extends PrivateCacheStrategy
+{
+    /**
+     * @var int
+     */
+    protected $ttl;
+
+    public function __construct(CacheStorageInterface $cache = null, $ttl)
+    {
+        $this->ttl = $ttl;
+
+        parent::__construct($cache);
+    }
+
+    protected function getCacheKey(RequestInterface $request)
+    {
+        return sha1(
+            'greedy'.$request->getMethod().$request->getUri()
+        );
+    }
+
+    public function cache(RequestInterface $request, ResponseInterface $response)
+    {
+        $response = $response->withAddedHeader(
+            'Warning',
+            'This response is cached although the response headers indicate not to do it!'
```

Comment:
Maybe here we can follow the RFC 7234 fromating? Like:
`299 - "This response is cached although the response headers indicate not to do it!"`
https://tools.ietf.org/html/rfc7234#section-5.5

What do you think @jeromegamez ?